src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -329,11 +329,20 @@ def to_pex_request(
         main: MainSpecification | None = None,
         sources: Digest | None = None,
     ) -> PexRequest:
+        if not interpreter_constraints:
+            if self.install_from_resolve and self.options.is_default("interpreter_constraints"):
+                # If installing the tool from a regular resolve, and custom ICs weren't
+                # explicitly set, leave these blank. This will cause the ones for the resolve
+                # to be used, which is clearly what the user intends, rather than forcing the
+                # user to override interpreter_constraints to match those of the resolve.
+                interpreter_constraints = InterpreterConstraints()
```

Comment:
We can't access options here, so we have to use blank ICs as a sentinel, and pick the right ones later.

Reply:
Your GH comment threw me since we definitely can access options here - plenty of methods do it above via `self`. I think you're referring to PythonSetup options, which we definitely don;t have here but need to perform the calculation.